draft-vandijk-dprive-ds-dot-signal-and-pin/draft-vandijk-dprive-ds-dot-signal-and-pin.md

```diff
@@ -94,6 +94,48 @@ The pseudo DNSKEY type can be used in CDNSKEY and CDS, as defined in RFC7344, re
 
 # Example
 
+This section will take you through the various parts of this specification, by example.
+
+We assume that we are working with a domain `example.com.` with one name server, `ns.example.com.`.
+
+## Generating and placing the (C)DNSKEY/DS records
+
+We will walk you through the CDNSKEY/DS generation, demonstrating it in terms of basic shell scripting and some common tools.
+
+First, we extract the SubjectPublicKeyInfo:
+
+```
+$ echo -n '' | openssl s_client -connect ns.example.com:853 | openssl x509 -noout -pubkey > pubkey.pem
+```
+
+This gives us a file `pubkey.pem` that looks like this (abridged):
+
+```
+-----BEGIN PUBLIC KEY-----
+MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAxH2a6NxIcw5527b04kKy
+...
+71AWASNoX2GQh7eaQPDD9i8CAwEAAQ==
+-----END PUBLIC KEY-----
+```
+
+To turns this into a CDNSKEY:
+
+1. remove the header and footer
+2. remove all newlines
```

Comment:
If you want a 1-liner that will extract the content:
```
openssl s_client -connect ns.example.com:853 </dev/null \
  | openssl x509 -noout -pubkey \
  | sed '1d;$d' \
  | tr -d '\n'
```

Reply:
something like that is in poc/README.me; I decided not to copy it here to make the procedure more explicit for the reader

Reply:
or perhaps we can do both, I'll add the code